AsyncDisplayKit/ASImageNode.mm

```diff
@@ -460,8 +464,12 @@ + (UIImage *)createContentsForkey:(ASImageNodeContentsKey *)key isCancelled:(asd
   // Another option is to have ASDisplayNode+AsyncDisplay coordinate these cases, and share the decoded buffer.
   // Details tracked in https://github.com/facebook/AsyncDisplayKit/issues/1068
   
-  @synchronized(key.image) {
-    [key.image drawInRect:key.imageDrawRect];
+  // We want to use copy, but if our image has alpha and our context is dirty, we have to use normal.
+  UIImage *image = key.image;
+  CGBlendMode blendMode = contextIsClean == NO && CGImageGetAlphaInfo(image.CGImage) != kCGImageAlphaNone ? kCGBlendModeNormal : kCGBlendModeCopy;
```

Comment:
I just wanted to avoid the alpha check if the context is clean. I'll clean it up.